pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go

```diff
@@ -40,21 +41,38 @@ import (
 )
 
 const (
-	projectID   = "test-project"
-	region      = "us-central1"
-	zoneName    = "us-central1-b"
-	nodeName    = "test-node-1"
-	clusterName = "Test Cluster Name"
-	clusterID   = "test-cluster-id"
-	serviceName = ""
+	projectID               = "test-project"
+	region                  = "us-central1"
+	zoneName                = "us-central1-b"
+	nodeName                = "test-node-1"
+	clusterName             = "Test Cluster Name"
+	clusterID               = "test-cluster-id"
+	serviceName             = "bar-svc"
+	eventReasonManualChange = "LoadBalancerManualChange"
+	eventMsgFirewallChange  = "Firewall change required by network admin"
+	errPrefixGetTargetPool  = "error getting load balancer's target pool:"
+	errStrLbNoHosts         = "Cannot EnsureLoadBalancer() with no hosts"
+	wrongTier               = "SupremeLuxury"
+	errStrUnsupportedTier   = "unsupported network tier: \"" + wrongTier + "\""
```

Comment:
Sorry, I should make it clearer. My point was if we hardcode these error messages in test, every time the actual code changes, the tests may need to change as well. In some cases it could be very confusing to later developers. I was thinking of making some of them constants and share among test and non-test codes.

Or maybe just check for the error type in test would be enough (if we properly type them...).

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce_loadbalancer_utils_test.go  could be a good place to put constants, if we want to share them between internal/external loadbalancer tests

Reply:
Moved to shared util. But still we need to formalize errors later for the sake of testing.